src/Server/HTTPHandler.cpp

```diff
@@ -500,7 +501,11 @@ bool HTTPHandler::authenticateUser(
     else if (request.getMethod() == HTTPServerRequest::HTTP_POST)
         http_method = ClientInfo::HTTPMethod::POST;
 
-    session->setHttpClientInfo(http_method, request.get("User-Agent", ""), request.get("Referer", ""));
+    NameValueCollection http_headers; 
+    for (const auto & it : request)
+        http_headers.set(it.first, it.second);
+
+    session->setHttpClientInfo(http_method, request.get("User-Agent", ""), request.get("Referer", ""), http_headers);
```

Comment:
request should be a `NameValueCollection`, can you simply send it here as argument?

Reply:
done